server/server.go

```diff
@@ -125,14 +126,23 @@ func initializeDefaultProject(opts ArgoCDServerOpts) error {
 	defaultProj := &v1alpha1.AppProject{
 		ObjectMeta: metav1.ObjectMeta{Name: common.DefaultAppProjectName, Namespace: opts.Namespace},
 		Spec: v1alpha1.AppProjectSpec{
-			SourceRepos:  []string{"*"},
-			Destinations: []v1alpha1.ApplicationDestination{{Server: "*", Namespace: "*"}},
+			SourceRepos:              []string{"*"},
+			Destinations:             []v1alpha1.ApplicationDestination{{Server: "*", Namespace: "*"}},
+			ClusterResourceWhiteList: []metav1.GroupKind{{Group: "*", Kind: "*"}},
 		},
 	}
 
-	_, err := opts.AppClientset.ArgoprojV1alpha1().AppProjects(opts.Namespace).Create(defaultProj)
-	if apierrors.IsAlreadyExists(err) {
-		return nil
+	existingDefault, err := opts.AppClientset.ArgoprojV1alpha1().AppProjects(opts.Namespace).Get(defaultProj.Name, metav1.GetOptions{})
+	if err == nil {
+		if !reflect.DeepEqual(existingDefault.Spec, defaultProj.Spec) {
+			existingDefault.Spec = defaultProj.Spec
+			_, err = opts.AppClientset.ArgoprojV1alpha1().AppProjects(opts.Namespace).Update(existingDefault)
+		}
```

Comment:
We want to allow users to customize the default project (just like any other project). So we don't want to replace the existing project if it already exists. For users upgrading from v0.8, we'll need need to put in a release note that cluster resources are disabled by default, and let them know the command to allow it.

Reply:
Good point. Reverted changes in server.go